jardin/instrumentation/instrumenter.py

```diff
@@ -0,0 +1,21 @@
+import time
+from .notifier import Notifer
+
+
+class Instrumenter:
+  def __init__(self, event_name, tags={}):
+      self.event_name = event_name
+      self.tags = tags
+      self.start_time = None
+      self.monotonic_start = None
+
+  def __enter__(self):
+      self.monotonic_start = time.monotonic()
+      self.start_time = time.time()
+
+  def __exit__(self, exc_type, exc_value, exc_traceback):
```

Comment:
Probably should differentiate between a call that finished successfully vs with an exception. In the latter case, the duration will be unusually short, skewing your metrics.

Try using the [@contextmanager decorator](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager). It makes these sorts of things a little easier to follow. I always forget how to use `__exit__()` correctly.

Reply:
I am okay with shorter durations as I plan to use this feature to tag spans and spans with errors are okay to cut short. I agree, though, that the subscriber might be interested in knowing about the failure.